### PR TITLE
Use NotificationChannelCompat.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Use `NotificationChannelCompat`.

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TacoTheDank** [comment]
AppCompat 1.2.0 to 1.3.0 is a very important library upgrade, being one of the most used core libraries of NewPipe. I'm of the opinion that this be thoroughly tested before being merged, at least to catch possible new behaviors and bugs before new GitHub issues are created.

AppCompat 1.3.0 changelog: https://developer.android.com/jetpack/androidx/releases/appcompat#version_130_3

**Comment by Redirion** [comment]
Thank you! I would like to merge all your recent prs, so also #6634, #6638, #6675, #6685 

apart from #6675 all PRs do increase the appcompat version. Could you do that in another PR seperately and rebase your PRs accordingly?

Also do you mind if the condense your PRs in the changelog like "general codequality updates" linking to your PRs?

**Comment by Isira-Seneviratne** [comment]
> Thank you! I would like to merge all your recent prs, so also #6634, #6638, #6675, #6685
> 
> apart from #6675 all PRs do increase the appcompat version. Could you do that in another PR seperately and rebase your PRs accordingly?
> 
> Also do you mind if the condense your PRs in the changelog like "general codequality updates" linking to your PRs?

Yeah, that's fine.

**Comment by Redirion** [comment]
> AppCompat 1.3.0 changelog: https://developer.android.com/jetpack/androidx/releases/appcompat#version_130_3

I was probablly a bit too quick in merging the appcompat lib update as I wanted to get that our of way to be able to merge these PRs.

The transitive dependencies are not matching. I guess we need to adopt.
androidxLifecycleVersion = '2.2.0' -> 2.3.1
'androidx.core:core-ktx:1.3.2' -> 1.5.0

**Comment by TacoTheDank** [comment]
> The transitive dependencies are not matching. I guess we need to adopt.
> androidxLifecycleVersion = '2.2.0' -> 2.3.1
> 'androidx.core:core-ktx:1.3.2' -> 1.5.0

Would be a good idea, yes 😉

File: app/src/main/java/org/schabi/newpipe/App.java [hunk 1]
```diff
@@ -233,38 +232,34 @@ protected void initACRA() {
     }
 
     private void initNotificationChannels() {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
-            return;
-        }
-
         String id = getString(R.string.notification_channel_id);
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/App.java:235]: Feel free to remove all of these `id` assignments and just inline the `getString()` call. It feels strange to reuse a variable this way, while inlining `getString()` is normal.

**Review by Stypox — CHANGES_REQUESTED** [review]

**Review by Stypox — APPROVED** [review]
Thank you! It works, both on API 19 emulated, API 29 real and API 30 emulated

**Pull request merged**